pilot/pkg/networking/core/v1alpha3/httproute_test.go

```diff
@@ -862,7 +862,6 @@ func TestSidecarOutboundHTTPRouteConfig(t *testing.T) {
 				"test-private.com:80": {
 					"test-private.com": true, "test-private.com:80": true, "9.9.9.9": true, "9.9.9.9:80": true,
 				},
-				"match-no-service.not-default:80": {"match-no-service.not-default": true, "match-no-service.not-default:80": true},
```

Comment:
@hzxuzhonghu This test case correctly fails when we select virtual services with matching services

Reply:
Why do we want this to change? I don't think a VS is required to match a service?

Reply:
https://github.com/istio/istio/pull/34247 - This change if VS host does not find a service, we drop it.

Reply:
The link is just to this PR, was it meant to be somewhere else?

I am a bit concerned we are making a backwards incompatible change.

Reply:
https://github.com/istio/istio/pull/33764 - Sorry this PR has changed that.

Reply:
Is that part actually merged?

> If you have concern of breaking some compatibility, we can leave 1, and probihit 2.

seems like it wasn't? Or I do not understand it :slightly_smiling_face:

Reply:
I am confused now. @hzxuzhonghu Is n't this expected if we select VS that only have services in the registry? Basically your fix in #33674 ?

Reply:
As i said for route 80, it is hacky, and there is a integration test that depend on the behavior. So i kept it.

Reply:
Ok. I fixed and this has not changed